src/connector/src/sink/encoder/template.rs

```diff
@@ -419,3 +434,213 @@ impl<T: SerTo<Vec<u8>>> SerTo<RedisSinkPayloadWriterInput> for T {
         ))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use risingwave_common::catalog::{Field, Schema};
+    use risingwave_common::row::OwnedRow;
+    use risingwave_common::types::{DataType, ScalarImpl};
+
+    use super::*;
+
+    #[test]
+    fn test_template_format_validation() {
+        // Create a schema with test columns
+        let schema = Schema::new(vec![
+            Field {
+                data_type: DataType::Int32,
+                name: "id".to_owned(),
+            },
+            Field {
+                data_type: DataType::Varchar,
+                name: "name".to_owned(),
+            },
+            Field {
+                data_type: DataType::Varchar,
+                name: "email".to_owned(),
+            },
+        ]);
+
+        // Create a map of column names to their data types
+        let mut map = HashMap::new();
+        for field in schema.fields() {
+            map.insert(field.name.clone(), field.data_type.clone());
+        }
+
+        // Test various template formats
+        let valid_templates = vec![
+            "user:{id}",
+            "user:\\{{id}",
+            "user:\\{{id}\\}",
+            "user:\\{{id},{name}\\}",
+            "user:\\{prefix{id},suffix{name}\\}",
+            "user:\\{prefix{id},suffix{name},email:{email}\\}",
+            "user:\\{nested\\{deeply{id}\\}\\}",
+            "user:\\{outer\\{inner{id}\\},another{name}\\}",
+            "user:\\{complex\\{structure\\{with{id}\\},and{name}\\},email:{email}\\}",
+            "user:{id}{name}",
+            "user:\\\\{id}",
+            "user:\\\\\\{id}",
+            "user:\\a{id}",
+            "user:\\b{name}",
+            "user:{id}{name}{email}",
+        ];
+
+        for template in valid_templates {
+            // Validate the template format
+            assert!(
+                TemplateStringEncoder::check_string_format(template, &map).is_ok(),
+                "Template '{}' should be valid",
+                template
+            );
+        }
+
+        // Test invalid templates
+        let invalid_templates = vec![
+            "user:no_braces",        // No braces
+            "user:{invalid_column}", // Non-existent column
+            "user:{id",              // Unclosed brace
+            "user:id}",              // Unopened brace
+            "sadsadsad{}qw4e2ewq21", // Empty braces
```

Comment:
Is `foo{id}bar{name` considered valid or invalid?
* if it is valid, the current description is wrong, as the unpaired `{` does not need escaping - only paired braces require escaping with backslash
* If it is invalid, the current implementation is wrongly accepting it - outputting `foo123bar{`

What about partially-escaped as in `foo\{id}bar` or `foo{id\}bar`?

As the template syntax gets more complicated, I would suggest finding an existing template library, rather than reinventing with our own buggy syntax and implementation. It maybe a breaking change, so better done earlier before it's too late.